airbyte-cdk/java/airbyte-cdk/core/src/test/java/io/airbyte/cdk/integrations/destination_async/state/GlobalAsyncStateManagerTest.java

```diff
@@ -314,4 +316,23 @@ private static long simulateIncomingRecords(final StreamDescriptor desc, final l
     return stateId;
   }
 
+  @Test
+  void flushingRecordsShouldNotReduceStatsCounterForGlobalState() {
+    final PartialAirbyteMessage globalState = new PartialAirbyteMessage()
+        .withState(new PartialAirbyteStateMessage().withType(AirbyteStateType.GLOBAL))
+        .withSerialized(Jsons.serialize(ImmutableMap.of("cursor", "1")))
+        .withType(Type.STATE);
+    final GlobalAsyncStateManager stateManager = new GlobalAsyncStateManager(new GlobalMemoryManager(TOTAL_QUEUES_MAX_SIZE_LIMIT_BYTES));
+
+    final long stateId = simulateIncomingRecords(STREAM1_DESC, 6, stateManager);
+    stateManager.decrement(stateId, 4);
+    stateManager.trackState(globalState, 1, STREAM1_DESC.getNamespace());
+    final List<PartialStateWithDestinationStats> stateBeforeAllRecordsAreFlushed = stateManager.flushStates();
+    assertEquals(0, stateBeforeAllRecordsAreFlushed.size());
+    stateManager.decrement(stateId, 2);
+    List<PartialStateWithDestinationStats> stateAfterAllRecordsAreFlushed = stateManager.flushStates();
+    assertEquals(1, stateAfterAllRecordsAreFlushed.size());
+    assertEquals(6.0, stateAfterAllRecordsAreFlushed.get(0).stats().getRecordCount());
```

Comment:
checking my understanding: without this PR, we would get 0 instead of 6 here?

Reply:
nope, we would get 2